File: Tree/ValidBST.py
# Create a function that return True if the Binary Tree is a BST or False if it isn't a BST

class Node:
	def __init__(self, val):
		self.val = val
		self.left = None
		self.right = None

def valid_BST(root):
	inorder = []
	def traverse(node):
		nonlocal inorder
		if not node:
			return
		traverse(node.left)
		inorder.append(node.val)
		traverse(node.right)

	traverse(root)
	if sorted(inorder) == inorder:
		return True
	else:
		return False

a = Node(5)
b = Node(2)
c = Node(7)
d = Node(4)
e = Node(9)

a.left = b
a.right = c
c.left = d
c.righ = e

print(valid_BST(a))

a = Node(5)
b = Node(2)
c = Node(7)
d = Node(4)
e = Node(9)

a.left = b
b.right = d
a.right = c
c.righ = e

print(valid_BST(a))

File: Tree/ValidBST.js
function Node(value){
	this.value = value;
	this.left = null;
	this.right = null;
}

function validBST(head){
	var nodes = [];
	function helper(node){
		if(!node){
			return;
		}
		helper(node.left);
		nodes.push(node.value);
		helper(node.right);
	}
	helper(head);
	for(var i=1; i<nodes.length; i++){
		if(nodes[i] < nodes[i-1]){
			return false;
		}
	}
	return true;
}

var a = new Node(20);
var b = new Node(10);
var c = new Node(30);
var d = new Node(5);
var e = new Node(7);

a.left = b;
a.right = c;
b.left = d;
d.right = e;

console.log(validBST(a)); 